components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/dao/impl/FeatureDAOImpl.java

```diff
@@ -261,6 +261,28 @@ public boolean deleteFeaturesOfProfile(Profile profile) throws FeatureManagerDAO
         }
     }
 
+    @Override
+    public boolean deleteFeaturesOfProfile(int profileId) throws FeatureManagerDAOException {
+        Connection conn;
+        PreparedStatement stmt = null;
```

Comment:
The 'stmt' variable should be initialized prior to passing to 'cleanupResources(stmt, resultset)' method